common/redis/templates/deployment.yaml

```
@@ -29,6 +29,12 @@ spec:
         {{- if and $.Values.global.linkerd_enabled $.Values.global.linkerd_requested }}
         linkerd.io/inject: enabled
         {{- end }}
+        {{- if .Values.config }}
+        checksum/configmap: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.redisPassword }}
+        checksum/secrets: {{ include (print .Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- end }}
```

I've looked through our other things and sometimes we use Template.BasePath and sometimes not. I leave it up to @majewsky what we use.

I don't thinkwe need to be conditional here, worst case we hash the empty files.

```suggestion
        checksum/configmap: {{ include "redis/templates/configmap.yaml" . | sha256sum }}
        checksum/secrets: {{ include "redis/templates/secrets.yaml" . | sha256sum }}
```

---

I believe the form with `.Template.BasePath` is necessary for library charts that are included as dependencies elsewhere, so the full path would be something like `my-app/charts/redis/templates/configmap.yaml`.